From 99d72aa195a3317abab409c41402124fc17212f0 Mon Sep 17 00:00:00 2001
From: KrishnaSimmadhari <91877513+KrishnaSimmadhari@users.noreply.github.com>
Date: Thu, 14 Dec 2023 21:36:17 +0530
Subject: [PATCH] mfd: intel-m10-bmc: Removing support for SDM with PMCI for
 CMC (#352)

Removing support for secure data manager for CMC as CMC does not support secure
updates using root key hashes.

Signed-off-by: Krishna Kumar S R <simmadhari.ramadass.krishna.kumar@intel.com>
---
 drivers/fpga/intel-m10-bmc-sec-update.c | 11 +++++++++++
 drivers/mfd/intel-m10-bmc-pmci.c        |  2 +-
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/fpga/intel-m10-bmc-sec-update.c b/drivers/fpga/intel-m10-bmc-sec-update.c
index 605574796617..bc67f642bfda 100644
--- a/drivers/fpga/intel-m10-bmc-sec-update.c
+++ b/drivers/fpga/intel-m10-bmc-sec-update.c
@@ -1590,6 +1590,13 @@ static const struct m10bmc_sec_ops m10sec_n6000_ops = {
 	.sec_visible = true,
 };
 
+static const struct m10bmc_sec_ops m10sec_cmc_ops = {
+	.rsu_status = m10bmc_sec_n6000_rsu_status,
+	.image_load = n6000_image_load_hndlrs,
+	.poc = &pmci_power_on_image,
+	.sec_visible = false,
+};
+
 #define SEC_UPDATE_LEN_MAX 32
 static int m10bmc_sec_probe(struct platform_device *pdev)
 {
@@ -1675,6 +1682,10 @@ static const struct platform_device_id intel_m10bmc_sec_ids[] = {
 		.name = "n6000bmc-sec-update",
 		.driver_data = (kernel_ulong_t)&m10sec_n6000_ops,
 	},
+	{
+		.name = "cmcbmc-sec-update",
+		.driver_data = (kernel_ulong_t)&m10sec_cmc_ops,
+	},
 	{ }
 };
 MODULE_DEVICE_TABLE(platform, intel_m10bmc_sec_ids);
diff --git a/drivers/mfd/intel-m10-bmc-pmci.c b/drivers/mfd/intel-m10-bmc-pmci.c
index 7272d35ef74b..f6df2bdc85c8 100644
--- a/drivers/mfd/intel-m10-bmc-pmci.c
+++ b/drivers/mfd/intel-m10-bmc-pmci.c
@@ -263,7 +263,7 @@ static struct mfd_cell m10bmc_pmci_c6100_bmc_subdevs[] = {
 
 static struct mfd_cell m10bmc_pmci_cmc_bmc_subdevs[] = {
 	{ .name = "cmcbmc-hwmon" },
-	{ .name = "n6000bmc-sec-update" },
+	{ .name = "cmcbmc-sec-update" },
 	{ .name = "cmcbmc-log" },
 };
 
